Do not Merge - Review of the complete codebase

josvandervelde opened the pull request:
This is a review of the complete codebase. 

The `empty-for-review` branch is empty. Once the review is complete, this PR can be closed and the `empty-for-review` branch can be deleted.

Taniya-Das committed:
Update README.md

Taniya-Das committed:
Create x

Taniya-Das committed:
Add files via upload

Taniya-Das committed:
Delete data/x

Taniya-Das committed:
Add files via upload

Taniya-Das committed:
Create x

Taniya-Das committed:
Add files via upload

Taniya-Das committed:
Delete scripts/x

Taniya-Das committed:
Update README.md

Taniya-Das committed:
Create methodology.md

Taniya-Das committed:
remove personal information

Taniya-Das committed:
use csv files instead of xlsx

Taniya-Das committed:
rename file

Taniya-Das committed:
correction in code to use local files

Taniya-Das committed:
code corrections

Taniya-Das committed:
clear output

Taniya-Das committed:
Update README.md

luisoala committed:
Delete scripts/openml_impact_analysis.ipynb

luisoala committed:
Create readme.md

luisoala committed:
Add files via upload

luisoala committed:
clear output

luisoala committed:
Delete scripts/openml_impact_analysis.ipynb

luisoala committed:
Rename openml_impact_analysis (1).ipynb to openml_impact_analysis.ipynb

luisoala committed:
Delete scripts/readme.md

PGijsbers committed:
add missing import, copy dataframe to avoid warning

PGijsbers committed:
Add a requirements file

PGijsbers committed:
Merge pull request #1 from luisoala/main

update code for figure 5

Taniya-Das committed:
add 71 newly colledted paper + script to add reviewer

Taniya-Das committed:
Update README.md

Taniya-Das committed:
script for correcting entry data

Taniya-Das committed:
update readme file

Taniya-Das committed:
update readme file

Taniya-Das committed:
add entry for paper 1695

Taniya-Das committed:
Merge pull request #3 from openml/add_entry_1695

add entry for paper 1695

Taniya-Das committed:
correct year for paper id 960

Taniya-Das committed:
correct year for paper id 960

Taniya-Das committed:
Merge pull request #4 from openml/correction_yar_id_960

correct year for paper id 960

PGijsbers committed:
Update Final_survey_data.csv

PGijsbers committed:
Update erroneous id

josvandervelde committed:
Set correct date for dataset id 56 (#7)

prabhant committed:
Update Final_survey_data.csv

Duplicate and wrong title corrected

Taniya-Das committed:
add new 67 entries to Final_survey_data

Taniya-Das committed:
Merge pull request #11 from openml/add_new_entries

add new 67 entries to Final_survey_data

SubhadityaMukherjee committed:
fix for duplicate data

SubhadityaMukherjee committed:
fix for duplicate data

SubhadityaMukherjee committed:
Merge pull request #12 from openml/subha_fix_new

subha fix new

janvanrijn committed:
update survey

Taniya-Das committed:
Merge pull request #13 from janvanrijn/main

update survey

Taniya-Das committed:
corrections in entries

Taniya-Das committed:
Merge pull request #15 from openml/corrections

corrections in entries

Taniya-Das committed:
update reuirements.txt

Taniya-Das committed:
correct df[paper year]

Taniya-Das committed:
add comments in code

Taniya-Das committed:
add pip install command

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
If it's in the current code, it is not a temporary measure.
Moreover, we might want to add some explanation regarding those date limits.

It might be better to show that currently, there are no papers between these years. Because in the final analysis, I hope we did not need to remove years before 2014 or after 2024, right?

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
I don't think we can drop any row without a good explanation. Why should this be done?

Taniya-Das replied:
This is because the paper is not available or behind paywall, so the paper year was not available/ left blank by the reviewer. I have double checked such entries.

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
This one is not empty. Why? How can there be before 2014? Some explanation would be helpful.

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
Why hardcode these values?

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
Naming: maybe `number_available_papers`?

josvandervelde replied:
@Taniya-Das this is still relevant

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
What is the hardcoded 1702 doing here?

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
Nit: "Number of" instead of "No fo"

josvandervelde replied:
@Taniya-Das this is still relevant

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
In the paper, 1719 and 1482 is mentioned :(

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
The values in the paper are slightly different!

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
Again, I get a slightly different value than the one in the paper

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
This 2017 needs some explanation

Taniya-Das replied:
because it was published in 2017.

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
Nit: papers instead of paper

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
Again, I see different values than the values in the paper

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
This seems unnecessary, they were filtered out already, right? Or am I missing something?

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
I really like this plot!

I see that at least 1 number is different in the code and in the paper: the number of 2024 papers, blue, is 219 in the paper, 221 in the code-plot.

Taniya-Das replied:
corrected

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
It would be nice to add an explanation: why are we filtering out 2025?

Taniya-Das replied:
"Note: Data from 2025 was excluded from this analysis as the year is still in progress while writing the paper, which could lead to misleading interpretations of yearly trends."  - added in figure caption. can't find a better place

josvandervelde reviewed (commented):
Overall, I did not find bugs. Mainly, I asked for some extra explanation. Also, and more severe, the numbers do not (yet?) correspond to the numbers in the paper.

Once it's ready for a second review, I'll try to check all the numbers in the paper, to check that I can reproduce the numbers using the notebook in this project

Taniya-Das committed:
corrections in data entry

Taniya-Das committed:
correct paper id 1385

Taniya-Das committed:
correct year

Taniya-Das committed:
convert str columns to lowercase and correct calculations

Taniya-Das committed:
Update README.md

Taniya-Das commented on docs/methodology.md:
```diff
@@ -0,0 +1,30 @@
+# Methodology
```
yes it's a duplicate. This was created here to verify the number from the notebook, and then was copied to the paper. If it feels unnecessary here, I can remove it.

josvandervelde replied:
I would suggest simplifying this file to 


```
# Methodology
See [paper appendix b](url/to/paper/appendix/b) 
```

To avoid differences between them.

Taniya-Das commented on scripts/additional_sanity_check.ipynb:
```diff
@@ -0,0 +1,297 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collected_papers = pd.read_csv(\"../data/collected_papers.csv\")\n",
```
The addiotnal check script was created for the second round to correct the entries received from first round. i think this file is not necessary here, as this script just for data correction. 

the sanity check is also in `openml_impact_analysis.ipyb` which I think is sufficient. 

Is it okay if I delete this?

josvandervelde replied:
Yes, I would be happy if this file was deleted, it felt duplicated indeed :)

josvandervelde replied:
Ah, I see you already deleted it. Good, I will click on "resolve" for the comments about this file

Taniya-Das committed:
correct image x-axis and create venv

Taniya-Das committed:
Delete scraped_paper_reviewer_assignment directory

Not directly related to impact analysis.

Taniya-Das committed:
Delete scripts/additional_sanity_check.ipynb

For internal checks for correcting the entries in the second round of review. Not necessary for sanity check analysis.

Taniya-Das committed:
Update README.md

Taniya-Das committed:
update variable name

Taniya-Das committed:
minor code corrections and naming conventions

Taniya-Das committed:
Update README.md

Taniya-Das committed:
Update methodology.md

Taniya-Das committed:
core_contibutor calculation

Taniya-Das committed:
delete image file

Taniya-Das committed:
update .gitignore file

Taniya-Das committed:
update readme.md

Taniya-Das committed:
strip jupyter notebook outputs.

Taniya-Das committed:
strip output from notebook

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
@Taniya-Das , this comment is still relevant

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
@Taniya-Das this comment is still relevant

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
```diff
@@ -0,0 +1,896 @@
+    "        })\n",
+    "        continue\n",
+    "\n",
+    "mismatches_df = pd.DataFrame(mismatches)\n",
```
nit: `df_mismatches` would be more normal naming

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
```diff
@@ -0,0 +1,896 @@
+    "\n",
+    "\n",
+    "out_of_range_2024 = df_survey_results[(df_survey_results[\"Paper Year\"] >2024)]\n",
+    "print(\"Papers with year after 2025:\", len(out_of_range_2024))\n",
```
Nit: this should be 2024, not 2025, 
and the next line (outcommented) can be removed

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
```diff
@@ -0,0 +1,896 @@
+    "\n",
+    "unvailable_papers = unvailable_papers + (1770-len(df_survey_results))\n",
+    "\n",
+    "# pepers with NaN or invalid entry type (not int) are unvalible (or being paywall)\n",
```
Nit: papers, not pepers

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
```diff
@@ -0,0 +1,896 @@
+    "# # Check the updated column type\n",
+    "# print(df_survey_results[\"Paper Year\"].dtypes)\n",
+    "\n",
+    "unvailable_papers = unvailable_papers + (1770-len(df_survey_results))\n",
```
`len(df_survey_results)` instead of 1770 (do not hardcode these values, it is not directly clear what they mean)a

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
```diff
@@ -0,0 +1,896 @@
+   "outputs": [],
+   "source": [
+    "Total_papers = pd.read_csv(path1) # Original list of scraped paper\n",
+    "print(\"Total number of scraped papers: \", len(Total_papers))"
```
When I run this, I get 1786. In the paper, it says "In total, we collected 1789 papers"

josvandervelde commented on scripts/openml_impact_analysis.ipynb:
```diff
@@ -0,0 +1,896 @@
+   },
+   "outputs": [],
+   "source": [
+    "# Datasets\n",
```
There is a lot of code duplication between this cell and most of the next cells, where the use-cases are counted. I'd suggest using this code instead, and removing all those cells:

```
df_papers_analyzed = available_papers  # I'd suggest renaming this variable beforehand


interaction_columns = [
    "Does the paper use datasets from OpenML?",
    "Does the paper use a collection (at least 2 or more) of datasets that are defined by OpenML designated to do benchmarking (e.g., openml benchmarking suites)?",
    "Does the paper use OpenML experiment data (i.e., utilise results from runs)? ",
    "Does the paper upload datasets to OpenML?",
    "Does the paper upload experiment data to OpenML?",
    "Does the paper interact in any other way with OpenML?"
]

use_case_filters = {
    "Total": df_papers_analyzed["Timestamp"].apply(lambda v: True),
    "Only citing OpenML": (df_papers_analyzed[interaction_columns] == "no").all(axis=1),
    "Use datasets": df_papers_analyzed["Does the paper use datasets from OpenML?"] == "yes",
    "Use benchmarking suites": df_papers_analyzed["Does the paper use a collection (at least 2 or more) of datasets that are defined by OpenML designated to do benchmarking (e.g., openml benchmarking suites)?"] == "yes",
    "Use experimental results": df_papers_analyzed["Does the paper use OpenML experiment data (i.e., utilise results from runs)? "] == "yes",
    "Upload datasets": df_papers_analyzed["Does the paper upload datasets to OpenML?"] == "yes",
    "Upload experimental data": df_papers_analyzed["Does the paper upload experiment data to OpenML?"] == "yes",
    "Other interactions": df_papers_analyzed["Does the paper interact in any other way with OpenML?"] == "yes",
    "Thesis publications": (df_papers_analyzed["Is the paper a thesis (Bachelor's, Master's, or PhD)?"] == "yes") | (df_papers_analyzed["Optional short description"].str.contains("thesis", case=False, na=False)),
    
}

use_cases = []
for name, use_case_filter in use_case_filters.items():
    count = df_papers_analyzed[use_case_filter].shape[0]
    percentage_of_papers = round(count / df_papers_analyzed.shape[0] * 100, 1)
    count_coauthored = df_papers_analyzed[(use_case_filter) & (df_papers_analyzed["Does the paper have at least 1 current OpenML Core Member as co-author?"] == "yes")].shape[0]
    percentage_coauthored = round(count_coauthored / count * 100, 1)
    use_cases.append({"Use case": name, "# of papers": count, "% of papers": percentage_of_papers, "% Co-authored by Core Members": percentage_coauthored})
pd.DataFrame(use_cases)
```

Taniya-Das committed:
correct count of paper
